src/audio/module_adapter/module_adapter.c

```diff
@@ -144,6 +145,91 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 	return NULL;
 }
 
+/**
+ * \brief Function to prepare module with input only
+ * @param dev component device pointer.
+ * @return 0 if succeeded, error code otherwise.
+ */
+static int only_input_module_prepare(struct comp_dev *dev)
```

Comment:
@iganakov would it be possible to use the simple_copy flag for the components of interest and modify the check for mod->num_input_buffers and mod->num_output_buffers to if (!mod->num_input_buffers && mod->num_output_buffers)?

Reply:
@ranj063 Ok, `simple_copy ` looks promising. I'm closing this PR and will open another one for simple copy implementation.